features/sharding/distsql/handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShowShardingKeyGenerateAlgorithmImplementationsExecutor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sharding.distsql.handler.query;
+
+import org.apache.shardingsphere.distsql.handler.ral.query.QueryableRALExecutor;
+import org.apache.shardingsphere.infra.merge.result.impl.local.LocalDataQueryResultRow;
+import org.apache.shardingsphere.infra.spi.ShardingSphereServiceLoader;
+import org.apache.shardingsphere.keygen.core.algorithm.KeyGenerateAlgorithm;
+import org.apache.shardingsphere.sharding.distsql.statement.ShowShardingKeyGenerateAlgorithmImplementationsStatement;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+
+/**
+ * Show sharding key generate algorithm implementations executor.
+ */
+public final class ShowShardingKeyGenerateAlgorithmImplementationsExecutor implements QueryableRALExecutor<ShowShardingKeyGenerateAlgorithmImplementationsStatement> {
+    
+    @Override
+    public Collection<String> getColumnNames() {
+        return Arrays.asList("name", "type", "class_path");
+    }
+    
+    @Override
+    public Collection<LocalDataQueryResultRow> getRows(final ShowShardingKeyGenerateAlgorithmImplementationsStatement sqlStatement) {
+        Collection<LocalDataQueryResultRow> result = new LinkedList<>();
+        Collection<KeyGenerateAlgorithm> shardingAlgorithms = ShardingSphereServiceLoader.getServiceInstances(KeyGenerateAlgorithm.class);
```

Comment:
`KeyGenerateAlgorithm` is now no longer in the sharding module (can also be used in other features), so can we remove the keyword `sharding`?

Reply:
> `KeyGenerateAlgorithm` is now no longer in the sharding module (can also be used in other features), so can we remove the keyword `sharding`?

Hi @RaigorJiang , the `SHOW SHARDING KEY GENERATE ALGORITHM IMPLEMENTATIONS` is in `shardingsphere-feature/shardingsphere-sharding` module, if we remove `sharding` key word, do we need to move to other module?

Reply:
Yes, since `KeyGenerateAlgorithm` is not in the sharding module, when the key word `sharding` removes, this DistSQL also needs to be moved outside of sharding, perhaps under the public parser module like `ShowComputeNodesStatement`.

`KeyGenerateAlgorithm` was moved out of the sharding module in #29066.

Reply:
I think this PR can be merged and the task of removing sharding will be left to the next PR.